js-core/homeworks/homework_6/src/main.js

```diff
@@ -0,0 +1,502 @@
+// uoY t'nod evah ot od gnihtyna laiceps ot nigeb gnisu eht .MOD tnereffiD sresworb evah tnereffid snoitatnemelpmi fo eht MOD
+
+console.log('The Document Object Model (DOM) is a programming interface for HTML and XML documents', '=> ', reverseEachWord('The Document Object Model (DOM) is a programming interface for HTML and XML documents'));
+// ehT tnemucoD tcejbO ledoM )MOD( si a gnimmargorp ecafretni rof LMTH dna LMX stnemucod
+
+/*
+ * TASK 3
+ * Добавьте в функцию reverseEachWord второй параметр,
+ * данный параметр булево, если true - тогда предложение так же
+ * переворачиваются в обратном порядке
+ * */
+console.log('\ntask 3:');
+
+
+console.log(reverseEachWord('You don\'t have to do anything special to begin using the DOM. Different browsers have different implementations of the DOM', true));
+//MOD eht fo snoitatnemelpmi tnereffid evah sresworb tnereffiD .MOD eht gnisu nigeb ot laiceps gnihtyna od ot evah t'nod uoY
+console.log(reverseEachWord('The Document Object Model (DOM) is a programming interface for HTML and XML documents', true));
+// stnemucod LMX dna LMTH rof ecafretni gnimmargorp a si )MOD( ledoM tcejbO tnemucoD ehT
+
+console.log(reverseEachWord('Hi my Name is', false));
+// iH ym emaN si
+
+/* TASK 4
+ * Посчитайте сколько одинаковых слов в предложении.
+ * Функция должна возвращать объект у которого свойства
+ * это повторяющиеся слово,
+ * а значение это количество повторений
+ * */
+
+// Both - Java - and - Java - Script - is - programming - and - programming - OOPBased
+
+function wordCounter(sentence) {
+  let obj =  {};
+  sentence.split(' ').forEach(function(value){
+    obj[value] = !obj[value] ? 1 : ++obj[value]; 
+  });
+  return obj
+}
+
+console.log('\ntask 4:');
+
+console.log(
+  wordCounter('Both Java and Java Script is programming and programming OOPBased Language'),
+);
+
+/*
+ {
+ Both:1,
+ Java:2,
+ and:2,
+ Script: 1,
+ is: 1
+ programming: 2
+ OOPBased:1,
+ Language:1
+ }
+ */
+
+console.log(wordCounter('asd qwe asd'));
+/*
+ {
+ asd:2
+ qwe:1
+ }
+ * */
+
+console.log(wordCounter('url http url www url http'));
+
+/*
+ {
+ http:2,
+ url:3,
+ www:1
+ }
+ * */
+
+/*
+  TASK 5
+ // Функция принимает массив у которого есть свойста _id и Company.
+ // верните объект, у которого ключ это _id, а значение Company
+ */
+
+function createHashTags(arr) {
+  
+  return arr.map(function(value){
+      let newObj = {};
+      newObj[value._id] = value.company;
+      return newObj;
+
+  });
+
+}
+console.log('\ntask 5:');
+
+console.log(createHashTags(listOfCompanys));
+//{"584babb6eeb4137cf14c37a3":"ASIMILINE", '584babb6eeb4137cf14c37a3':'Company2', }
+
+
+
+console.log('\ntask super 1:');
+
+// @ SUPER
+/*
+ *
+ * TASK 1
+ * Выведите уникальные значения
+ *
+ * */
+
+function uniqueElements(arr) {
+  let _tempObj = {};
+  let _resArr = []
```

Comment:
I guess we can improve your solution. 
And your solution won't work for the first array.

Can we solve it with one loop?